catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -150,20 +140,130 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
                     .build();
         }
 
-        Map<String, Object> registryMap = reportMapBuilder.buildRegistryMap(registryPackage);
+        try {
+            html = registryReportBuilder.getHtmlFromRegistryPackage(registryPackage, "report");
+        } catch (IOException e) {
+            String message = "Error in compiling and applying report template.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        return Response.ok(html)
+                .build();
+    }
+
+    @Path("/{registryId}/report/summary")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewSummaryInfoHtml(@PathParam("registryId") final String registryId) {
+        String html = "";
+        Metacard metacard;
+
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+        try {
+            metacard =
+                    (federationAdminService.getRegistryMetacardsByRegistryIds((Collections.singletonList(
+                            registryId)))
```

Comment:
As long as there is a value in registryId, there should be an element. I will double check with the Blue team though!